src/v/cloud_storage/cache_service.cc

```diff
@@ -885,29 +838,66 @@ cache::reserve_space(uint64_t bytes, size_t objects) {
     co_return space_reservation_guard(*this, bytes, objects);
 }
 
-void cache::reserve_space_release(uint64_t bytes, size_t objects) {
+void cache::reserve_space_release(
+  uint64_t bytes,
+  size_t objects,
+  uint64_t wrote_bytes,
+  uint64_t wrote_objects) {
     vlog(
       cst_log.trace,
-      "reserve_space_release: releasing {}/{} reserved bytes/objects",
+      "reserve_space_release: releasing {}/{} reserved bytes/objects (wrote "
+      "{}/{})",
       bytes,
-      objects);
+      objects,
+      wrote_bytes,
+      wrote_objects);
 
     if (ss::this_shard_id() == ss::shard_id{0}) {
-        do_reserve_space_release(bytes, objects);
+        do_reserve_space_release(bytes, objects, wrote_bytes, wrote_objects);
     } else {
-        ssx::spawn_with_gate(_gate, [this, bytes, objects]() {
-            return container().invoke_on(0, [bytes, objects](cache& c) {
-                return c.do_reserve_space_release(bytes, objects);
-            });
-        });
+        ssx::spawn_with_gate(
+          _gate, [this, bytes, objects, wrote_bytes, wrote_objects]() {
+              return container().invoke_on(
+                0, [bytes, objects, wrote_bytes, wrote_objects](cache& c) {
+                    return c.do_reserve_space_release(
+                      bytes, objects, wrote_bytes, wrote_objects);
+                });
+          });
     }
 }
 
-void cache::do_reserve_space_release(uint64_t bytes, size_t objects) {
+void cache::do_reserve_space_release(
+  uint64_t bytes, size_t objects, uint64_t wrote_bytes, size_t wrote_objects) {
     vassert(ss::this_shard_id() == ss::shard_id{0}, "Only call on shard 0");
     vassert(_reserved_cache_size >= bytes, "Double free of reserved bytes?");
     _reserved_cache_size -= bytes;
     _reserved_cache_objects -= objects;
+
+    _current_cache_size += wrote_bytes;
+    _current_cache_objects += wrote_objects;
+    probe.set_size(_current_cache_size);
+    probe.set_num_files(_current_cache_objects);
+    if (
+      _current_cache_size > _max_bytes()
+      || _current_cache_objects > _max_objects()) {
+        // This should not happen, because callers to put() should have used
+        // reserve_space() to ensure they stay within the cache size limit. This
+        // is not a fatal error in itself, so we do not assert, but emitting as
+        // an ERROR log ensures detection if we hit this path in our integration
+        // tests.
+        vlog(
+          cst_log.warn,
```

Comment:
Meant to be error, per the comment?

Reply:
This was already a WARN prior to this PR, but now that cache size enforcement is even more strict, I think we can upgrade it to an ERROR.  It'll be interesting to see if this fails any other existing tests.